### PR TITLE
Cookiesync: Refactor configuration

### PR DESCRIPTION
**Pull request by And1sS**
This pr will make cookie-sync configuration to be more in-line with GO version.

Example of usersync configuration this pr introduces:
```yaml
    usersync:
      cookie-family-name: bidder_name
      redirect:
        url: bidder_url?gdpr={{gdpr}}&gdpr_consent={{gdpr_consent}}&us_privacy={{us_privacy}}&redir={{redirect_url}}
        support-cors: false
        uid-macro: '[BIDDER_UID_MACRO]'
        format-override: blank
      iframe:
        url: bidder_url?gdpr={{gdpr}}&gdpr_consent={{gdpr_consent}}&us_privacy={{us_privacy}}&redir={{redirect_url}}
        support-cors: false
        uid-macro: '[BIDDER_UID_MACRO]'
        format-override: pixel
```
fields `uid-macro` and `format-override` can be omitted, as well as `iframe`, `redirect` sections or even whole `usersync` section.
More detailes in docs pr: https://github.com/prebid/prebid.github.io/pull/3916

### CODE REVIEW

File: src/main/java/org/prebid/server/auction/ExchangeService.java [hunk 1]
```diff
@@ -751,7 +751,9 @@ private String extractUid(Map<String, String> uidsBody, UidsCookie uidsCookie, S
      * Extract cookie family name from bidder's {@link Usersyncer} if it is enabled. If not - return null.
      */
     private String resolveCookieFamilyName(String bidder) {
-        return bidderCatalog.isActive(bidder) ? bidderCatalog.usersyncerByName(bidder).getCookieFamilyName() : null;
+        return bidderCatalog.isActive(bidder)
+                ? ObjectUtil.getIfNotNull(bidderCatalog.usersyncerByName(bidder), Usersyncer::getCookieFamilyName)
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/auction/ExchangeService.java:755]: Just to think. Maybe it is worth making `usersyncerByName` return Optional?

File: src/main/java/org/prebid/server/bidder/UsersyncFormat.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    @JsonProperty("pixel")
+    PIXEL("i"),
+
+    @JsonProperty("blink")
+    BLINK("b");
+
+    public final String name;
+
+    UsersyncFormat(String name) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/UsersyncFormat.java:15]: Can we delete this constructor and add `@RequiredArgsConstructor`?

File: src/main/java/org/prebid/server/bidder/UsersyncInfoBuilder.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package org.prebid.server.bidder;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.privacy.model.Privacy;
+import org.prebid.server.proto.response.UsersyncInfo;
+import org.prebid.server.util.HttpUtil;
+
+public class UsersyncInfoBuilder {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/UsersyncInfoBuilder.java:9]: Just to think. Can this one be `UsersyncInfoResolver`? Then it is not required to use this complex and useless for this case builder approach(whole class will be refactored)?

File: src/main/java/org/prebid/server/bidder/UsersyncMethodType.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    IFRAME("iframe", UsersyncFormat.BLINK),
+    REDIRECT("redirect", UsersyncFormat.PIXEL);
+
+    public final String name;
+    public final UsersyncFormat format;
+
+    UsersyncMethodType(String name, UsersyncFormat format) {
+        this.name = name;
+        this.format = format;
+    }
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/UsersyncMethodType.java:14]: Maybe `@RequiredArgsConstructor`?

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

**Review by Net-burst — APPROVED** [review]
LGTM

**Review by SerhiiNahornyi — APPROVED** [review]

**Comment by richaudience** [comment]
Hi Guys,
I see that the last Update was to 4 months, could you please tell me for when this programmed the next refactor for cookie-sync?
I need that upgrade the richaudience .yaml file
Cheers

**Comment by And1sS** [comment]
> Hi Guys, I see that the last Update was to 4 months, could you please tell me for when this programmed the next refactor for cookie-sync? I need that upgrade the richaudience .yaml file Cheers

Hi, which update are you talking about? We have plenty of them planned :)

**Comment by richaudience** [comment]
Hi,
I am talking specifically about the richaudience adapter, we only have to add the image sync to our .yalm file. That is possible?
Cheers

**Comment by richaudience** [comment]
Hi guys!
Have been able to check my last comment?? I wait for your response.
Cheers

**Pull request merged**